src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -106,7 +106,19 @@ private function parseImports(\DOMDocument $xml, $file)
         $defaultDirectory = dirname($file);
         foreach ($imports as $import) {
             $this->setCurrentDir($defaultDirectory);
-            $this->import($import->getAttribute('resource'), XmlUtils::phpize($import->getAttribute('type')) ?: null, (bool) XmlUtils::phpize($import->getAttribute('ignore-errors')), $file);
+
+            $resourceFile = $import->getAttribute('resource');
+            if (!$this->isAbsolutePath($resourceFile)) {
+                $resourceFile = $defaultDirectory.DIRECTORY_SEPARATOR.$resourceFile;
+            }
+
+            if (array() !== $files = glob($resourceFile, defined('GLOB_BRACE') ? GLOB_BRACE : 0)) {
+                foreach ($files as $resource) {
+                    $this->import($resource, XmlUtils::phpize($import->getAttribute('type')) ?: null, (bool) XmlUtils::phpize($import->getAttribute('ignore-errors')), $file);
```

Comment:
The value of 'type' and 'ignore-errors' will not change within this loop, so it's better to resolve them before the loop, and then reuse the value.